block/async_store.go

```diff
@@ -0,0 +1,45 @@
+package block
+
+import (
+	"context"
+	"time"
+
+	"github.com/rollkit/rollkit/pkg/store"
+)
+
+const DefaultFlushInterval = 1 * time.Second
+
+// AsyncPruner is a service that periodically prunes block data in the background.
+type AsyncPruner struct {
+	ps store.PruningStore
+
+	flushInterval time.Duration
+}
+
+func NewAsyncPruner(pruningStore store.PruningStore, flushInterval time.Duration) *AsyncPruner {
+	if flushInterval <= 0 {
+		flushInterval = DefaultFlushInterval
```

Comment:
personal preference: IMHO it is fair to return an error for invalid values. I assume `0` is just using default